subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -107,12 +107,23 @@ void lll_adv_prepare(void *param)
 {
 	struct lll_prepare_param *p = param;
 	int err;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
```

Comment:
You do not need to implement the event counting in this PR for generation of adv set terminate event, because both Max_Extended_Advertising_Events parameter and duration parameter to HCI_LE_Set_Extended_Advertising_Enable command is not yet implemented in the controller.

From the BT HCI specification:
`If the Max_Extended_Advertising_Events parameter in the
HCI_LE_Set_Extended_Advertising_Enable command was non-zero, the
Num_Completed_Extended_Advertising_Events parameter shall be set to the
number of completed extended advertising events the Controller had
transmitted when either the duration elapsed or the maximum number of
extended advertising events was reached; otherwise it shall be set to zero.`